### ISSUE TITLE
PUT /api/v1/users silently ignores empty roles array — cannot remove all roles from a user

### ISSUE THREAD

**Issue by hassandotcms** [dotCMS/core#37109]
### Problem Statement

`PUT /api/v1/users` accepts a `roles` array whose contract (established in #33410) is full replacement: the list sent becomes the user's complete role set.

This works for every input except the empty array. Sending `"roles": []` — the admin's way of saying "this user should hold no roles" — returns **200 OK but changes nothing**. The empty list is treated as if the field was not sent.

The legacy Users portlet supports this operation: un-assigning every role and saving calls DWR `UserAjax.updateUserRoles(userId, [])`, which removes all of the user's user-assignable roles. The REST API cannot do what the legacy UI can, and it fails silently — the caller gets a success response while the user keeps CMS Administrator, Back-end User, Front-end User, etc.

**Root cause:** `UserResource#processRoles` gates on `UtilMethods.isSet(roles)`, which is `false` for both `null` and an empty list; additionally `UserForm` normalizes a missing `roles` field to an empty list at construction, erasing the distinction between "not sent" and "sent empty" before the resource can act on it.

**Expected behavior:**

- `roles` **absent** from the payload → roles untouched (current behavior, unchanged).
- `roles` **non-empty** → full replacement (current behavior, unchanged).
- `roles: []` → all of the user's **user-assignable** roles (`editUsers = true`) are removed, matching legacy `UserAjax.updateUserRoles` semantics. System-managed roles that role editing never touches (e.g. the user's individual role) are unaffected, exactly as in the legacy portlet.

### Consistency: apply the same rule to the role-side membership endpoint

Both membership-mutation surfaces must enforce the same legacy rule — only
user-assignable roles (`editUsers = true`) are manageable:

- `POST /v1/roles/{roleId}/users/{userId}` (grant) already returns **403** for
  non-assignable roles.
- `DELETE /v1/roles/{roleId}/users` (bulk remove) currently has no such gate: it
  accepts a non-assignable role id (e.g. a user's individual role, discoverable via
  `GET /v1/roles/users/{userIdOrEmail}`) and unlinks the membership. The legacy
  Roles portlet blocks this in its UI: for a role with `editUsers = false` the user
  rows render no selection checkboxes and the Remove button is hidden
  (`view_roles_js_inc.jsp`), so this removal was never reachable end-to-end.

This endpoint merged in #37077 on Aug 18, 2026 and has only shipped in the agile
releases cut since (v26.08.19-03+); it is not in any LTS and predates any client
adoption (the Angular Roles portlet that will consume it is not built yet), so
aligning it now is effectively non-breaking. The gate mirrors the grant endpoint:
**403 when `role.isEditUsers()` is false** (role-level, whole request — not a
per-user `skipped` reason, since assignability is a property of the role).

### Steps to Reproduce

1. Create a user with roles:

```
curl -X POST '/api/v1/users' -H 'Content-Type: application/json' -d '{
  "firstName": "Test", "lastName": "User", "email": "roles-test@dotcms.com",
  "password": ["1","2","3","4","5","6","7","8"],
  "roles": ["DOTCMS_BACK_END_USER", "Scripting Developer"]
}'
```

2. Update the user with an empty roles array:

```
curl -X PUT '/api/v1/users' -H 'Content-Type: application/json' -d '{
  "userId": "<userId from step 1>",
  "firstName": "Test", "lastName": "User", "email": "roles-test@dotcms.com",
  "roles": []
}'
```

3. Load the user's roles (e.g. `GET /api/v1/roles/users/<userId>`).

**Expected:** the user holds no user-assignable roles.
**Actual:** 200 OK, the user still holds Back-end User and Scripting Developer.

### Acceptance Criteria

- [ ] `PUT /api/v1/users` with `"roles": []` removes all user-assignable roles from the user
- [ ] `PUT /api/v1/users` without a `roles` field leaves roles untouched
- [ ] Non-empty `roles` behavior is unchanged
- [ ] OpenAPI description of the `roles` field documents all three cases
- [ ] Integration/Postman coverage asserts the resulting role set for each case
- [ ] `DELETE /v1/roles/{roleId}/users` returns 403 when the target role is not user-assignable (`editUsers = false`), mirroring the grant endpoint
- [ ] OpenAPI description of the DELETE endpoint documents the 403 and defines "user-assignable"

### dotCMS Version

Latest from main branch (behavior introduced with the #33410 fix, PR #33414)

### Severity

Medium - Some functionality impacted

### Links

NA

**Comment by hmoreras** [comment]
## QA — PR #37156 verified ✅

  **Result:** 29 requests, 82 assertions, 0 failures.

  **Environment:**
  - Build: `dotcms/dotcms:modernization` image, built 2026-08-31 (contains PR #37156, merged 2026-08-22)
  - Server: `http://localhost:8081` (local push-publish sender)
  - Runner: newman 6.2.1
  - Date: 2026-09-02

  **Endpoints exercised** (one throwaway user, created and deleted by the collection):
  - `PUT /v1/users` `"roles": []` — **AC1**: all user-assignable roles removed; the user's individual role (`editUsers=false`) preserved
  - `PUT /v1/users` **roles field absent** — **AC2**: role set unchanged, while a `firstName` change in the same payload is applied (so "untouched" isn't a silent no-op)
  - `PUT /v1/users` **non-empty roles** — **AC3**: full replacement; dropped role loses its direct membership, kept role retains it, previously-held `DOTCMS_BACK_END_USER`
  removed
  - `PUT /v1/users` with a **null** and a **blank** entry — 400 each, and rollback verified in both directions (the valid entry in the rejected payload was not granted; the
  existing membership was not revoked)
  - `PUT /v1/users` with an **unknown key + a non-assignable key** — 200, both ignored, nothing granted (regression: a non-assignable key used to abort the request with
  `DotStateException`)
  - `DELETE /v1/roles/{roleid}/users` — **AC6**: 403 on a non-user-assignable role, membership intact afterwards; plus a positive control confirming removal still works on an
  assignable role
  - `GET /api/openapi.json` — **AC4** and **AC7** asserted against the spec the server actually serves: the `updateUser` description covers absent / empty / non-empty and
  defines `editUsers`; the `removeUsersFromRole` 403 is declared and documents the flag
